### DISCUSSION TITLE
Why can't both CollaboraOffice and LibreOffice be enabled at the same time?

### DISCUSSION THREAD

**Discussion by AlphaCraft9658** [nextcloud/all-in-one#5656] [Questions] [answered]
OnlyOffice has been added as an option in Nextcloud-AIO fairly recently, from what I can tell. I wanted to enable it there, so I don't have to use an external OnlyOffice-documentserver and because some of the issues might not exist in the integrated version. But I got the message that CollaboraOffice and OnlyOffice cannot be used at the same time. I can use it with the OnlyOffice app when also having Collabora. Why can't I enable the integrated options for both at the same time? I couldn't find any information on this. Does anybody know something?

**Comment by szaimen** [comment, marked as answer]
Hi, this has technical reasons as they are incompatible with each other. For more questions, please ask at https://help.nextcloud.com. Thanks!

**Reply by AlphaCraft9658** [reply]
Because we are here already, I hope it's okay to ask here. Is it some kind of internal AIO issue with hosting both at the same time or is it about using both inside of Nextcloud at the same time? The latter at least doesn't appear to be a problem for me, as I can still use both the integrated CollaboraOffice and set up an external OnlyOffice integration at the same time. I would like to further understand the technical implications.

**Comment by neocult-de** [comment]
It is an unsupported setup, but we used it successfully so far anyways since Nextcloud 22 up to current 30. The restriction can be gone around by directly editing the master configuration file and set a 1 to OnlyOffice nonetheless.

We set the Default association of files for Microsoft Formats docx, xlsx etc. to OnlyOffice and keep everything else on Nextcloud Office. Should be mutually exclusive!

We have a cronjob running to make sure the documentserver is properly recognized after the nightly backup:
0 5 * * * sudo docker exec --user www-data -it nextcloud-aio-nextcloud php occ onlyoffice:documentserver --check

In that way we open the open formats with Nextcloud Office and the closed formats with OnlyOffice so far reliably ... But as noted: Unsupported!!

**Reply by AlphaCraft9658** [reply]
> It is an unsupported setup, but we used it successfully so far anyways since Nextcloud 22 up to current 30. The restriction can be gone around by directly editing the master configuration file and set a 1 to OnlyOffice nonetheless.
> 
> We set the Default association of files for Microsoft Formats docx, xlsx etc. to OnlyOffice and keep everything else on Nextcloud Office. Should be mutually exclusive!
> 
> We have a cronjob running to make sure the documentserver is properly recognized after the nightly backup:
> 0 5 * * * sudo docker exec --user www-data -it nextcloud-aio-nextcloud php occ onlyoffice:documentserver --check
> 
> In that way we open the open formats with Nextcloud Office and the closed formats with OnlyOffice so far reliably ... But as noted: Unsupported!! 

I am wondering why it's unsupported then. I mean, if it works it works. I could either use the integrated document-server or an external one, although I'd prefer the first option, as I would have less endpoints to directly expose to the internet. How would I change the value for the master configuration file though? Or even better: Where is that file and what value would it be exactly?

**Reply by neocult-de** [reply]
The mastercontainer configuration can be accessed as writen in the documentation:

`sudo docker run -it --rm --volume nextcloud_aio_mastercontainer:/mnt/docker-aio-config:rw alpine sh -c "apk add --no-cache nano && nano /mnt/docker-aio-config/data/configuration.json"`

There is a section with the containers:
"isClamavEnabled": 1,
    "isOnlyofficeEnabled": 1,
    "isCollaboraEnabled": 1,
    "isTalkEnabled": 0,
    "isTalkRecordingEnabled": 0,
    "isImaginaryEnabled": 1,
    "isFulltextsearchEnabled": 1,

whereby 1 is enabled, 0 is disabled as you would do it in the Mastercontainer web interface. I always have activated Collabora/Nextcloud office and the other containers I need via the webinterface. Then go this configuration file while the containers are stopped, change "isOnlyofficeEnabled": 1, (from 0 to 1), save it to the file and then start Nextcloud-AIO via the webinterface again.

---
The issue is that both programs try to be your default solution for opening documents and behaviour which one does what, with which priority is undefined. I also needed to add the cronjob to check for occ onlyoffice:documentserver --check  after a backup or update or only office file associations wouldn´t work.

But if configured the Fileformats in the Administration Settings within the Cloud under Nextcloud Office and OnlyOffice respectively that they react mutually exclusive to the different file formats it works great so far for me. And in the context menu of files ( ... / Three Dot Menu) you can choose "Open with" either of the two office solutions.

I had this setup running before Nextcloud-AIO as well, as I would recommend open formats and Nextcloud Office to everyone, but some institutions insist on Microsoft Formats and then OnlyOffice is actually doing a better job on these few files. But for a few files I also don´t want to abandon the open standards and from my point of view overall better product.
